### PR TITLE
feat(OCPADVISOR-71): Show upgrade risks table

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/OCPADVISOR-71.
Mocked at https://www.sketch.com/s/c5c07e06-f4a1-4c64-9cdf-8b12d8ac9ad5/a/25OY5ve.

This introduces the upgrade risks table that is shown under the "Upgrade risks" tab on the cluster details view. 

- The data for this table are fetched from the new back-end (Smart proxy) endpoint `v1/cluster/%id/upgrade-risks-prediction`.
- The table must be shown if there are firing alerts or cluster operators available.
- When there are no firing alerts nor cluster operators, the tab must show successful empty message.
- If the response from back end is 503, then the tab must show the "not available" empty state.
- For all other edge cases that are unknown/unexpected, the tab must show "something went wrong."  

## How to test

The real endpoint will serve no data for real clusters, thus it is better to run this PR together with the mock service: 
1. Follow https://github.com/RedHatInsights/ocp-advisor-frontend#using-insights-results-aggregator-mock instructions and make sure the webpack server and the mock server are running.
2. Use different cluster IDs from https://github.com/RedHatInsights/insights-results-aggregator-mock/#clusters-that-return-valid-data and navigate to `https://stage.foo.redhat.com:1337/beta/openshift/insights/advisor/clusters/%ID`.
3. Click on the "Upgrade risks" tab.
4. Make sure the content matches with the specifics of this cluster (e.g., `00000003-eeee-eeee-eeee-000000000001` must show firing alerts and cluster operators).

## Screenshots

image

image

image

image

image

### CODE REVIEW

File: src/Services/SmartProxy.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ export const SmartProxyApi = createApi({
       query: ({ id }) => `v2/cluster/${id}/info`,
       transformResponse: (response) => response?.cluster,
     }),
+    getUpgradeRisks: builder.query({
+      query: ({ id }) => `v1/cluster/${id}/upgrade-risks-prediction`,
```
Comment (ikerreyes) [inline src/Services/SmartProxy.js:41]: I think upgrade risk predictions is under v2

Comment (juandspy) [reply src/Services/SmartProxy.js:41]: it is

**Review by ikerreyes — APPROVED** [review]
Thanks a lot, great work with the tests

File: src/Components/UpgradeRisksTable/UpgradeRisksTable.cy.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
+import React from 'react';
+import upgradeRisksFixtures from '../../../cypress/fixtures/api/insights-results-aggregator/v1/clusters/41c30565-b4c9-49f2-a4ce-3277ad22b258/upgrade-risks-prediction.json';
+import { TABLE_HEADER } from '../../../cypress/utils/components';
+import {
+  checkEmptyState,
+  checkTableHeaders,
+} from '../../../cypress/utils/table';
+import UpgradeRisksTable from './UpgradeRisksTable';
+
+const SEVERITY_MAPPING = {
```
[the remark this answers is not in the document]
Comment (gkarat) [reply src/Components/UpgradeRisksTable/UpgradeRisksTable.cy.js:10]: I wanted to keep it separately so that tests can spot a change and not be automatically updated together with implementation

**Review by adonispuente — APPROVED** [review]
Great work, didnt find any issues, LGTM!

**Comment by Fewwy** [comment]
After the update for the mock server, I see every type of table that is on the mocks. Great job. LGTM

**Review by Fewwy — APPROVED** [review]

File: src/Components/UpgradeRisksTable/AlertsList.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+export const ALERTS_SEVERITY_ORDER = ['critical', 'warning', 'info'];
+
+const AlertsList = () => {
+  const { clusterId } = useParams();
+  const { data } = useGetUpgradeRisksQuery({ id: clusterId });
+  const { alerts = [] } =
+    data?.upgrade_recommendation?.upgrade_risks_predictors || {};
+
+  return (
+    <TableComposable
```
Comment (adonispuente) [inline src/Components/UpgradeRisksTable/AlertsList.js:62]: When comparing to the mocks I see a few links/icons that arent shown here.
Your version:
Screen Shot 2023-04-14 at 9 16 19 AM

mock version:
Screen Shot 2023-04-14 at 9 17 15 AM

Comment (gkarat) [reply src/Components/UpgradeRisksTable/AlertsList.js:62]: good catch. there is no final agreement on the docs URL, so I'd wait for folks first to decide and then will add it via separate PR most probably
